src/data/queries/reactWidget.ts

```diff
@@ -0,0 +1,23 @@
+import { ParagraphReactWidget } from '@/types/drupal/paragraph'
+import { QueryFormatter } from 'next-drupal-query'
+import { ReactWidget } from '@/types/formatted/reactWidget'
+
+export const formatter: QueryFormatter<ParagraphReactWidget, ReactWidget> = (
+  entity: ParagraphReactWidget
+) => {
+  return {
```

Comment:
not a blocker just a stylistic question. all of these have `entity?.` in some fashion, could we just `if (!entity) return` beforehand?

Reply:
The returned object needs to comply with the type `ReactWidget` which requires `entityId` and `widgetType`. So, we can't just return an empty object.

That said, this did make me realize that I don't need all those `?` instances. `entity` will always be defined, so those are not necessary.

With regards to the repeated `entity.[some_field]`, there are other ways we could write this to not have that repeated everywhere, but they don't necessarily make the code any more concise (see example below).

```
  const {
    drupal_internal__id,
    field_widget_type,
    field_cta_widget,
    field_loading_message,
    field_timeout,
    field_error_message,
    field_default_link,
    field_button_format,
  } = entity

  return {
    entityId: drupal_internal__id,
    widgetType: field_widget_type,
    ctaWidget: field_cta_widget,
    loadingMessage: field_loading_message,
    timeout: field_timeout,
    errorMessage: field_error_message.processed,
    defaultLink: field_default_link
      ? {
          url: field_default_link.url,
          title: field_default_link.title,
        }
      : null,
    buttonFormat: field_button_format,
  }
```

Reply:
yeah not worried at all about deconstructing off for the formatter, was mostly just curious about the `?`. if reactwidget always needs it / will always have it, cool